### PR TITLE
Move HTTP-related code into a separate crate

### PR DESCRIPTION
**Pull request by djc**
See #403.

### CODE REVIEW

**Review by jtescher — APPROVED** [review]
LGTM

**Comment by jtescher** [comment]
Can add the usual crate files license, readme, etc as well or in a follow up pr

File: opentelemetry-http/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    #[derive(Debug)]
+    struct SurfError(surf::Error);
+
+    impl ExportError for SurfError {
+        fn exporter_name(&self) -> &'static str {
+            "surf"
+        }
+    }
+
+    #[cfg(feature = "surf")]
```
Comment (TommyCpp) [inline opentelemetry-http/src/lib.rs:133]: Do we still need this even the mod is guarded by a `#[cfg(feature = "surf")]`?

Comment (djc) [reply opentelemetry-http/src/lib.rs]: We don't, good catch! Fixed it in the latest version.

**Review by TommyCpp — APPROVED** [review]
Nice work! Thanks a lot

**Pull request merged**